src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -649,11 +648,38 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
                             triggerFlags = TRIGGERED_FULL_MASK;
                     }
 
-                    me->CastSpell(target->ToUnit(), e.action.cast.spell, triggerFlags);
+                    SpellCastResult result = me->CastSpell(target->ToUnit(), e.action.cast.spell, triggerFlags);
+                    bool spellCastFailed = (result != SPELL_CAST_OK && result != SPELL_FAILED_SPELL_IN_PROGRESS);
+
+                    if (e.action.cast.castFlags & SMARTCAST_COMBAT_MOVE)
+                    {
+                        // If cast flag SMARTCAST_COMBAT_MOVE is set combat movement will not be allowed unless target is outside spell range, out of mana, or LOS.
+                        if (result == SPELL_FAILED_OUT_OF_RANGE)
+                            // if we are just out of range, we only chase until we are back in spell range.
+                            CAST_AI(SmartAI, me->AI())->SetCombatMove(true, spellMaxRange - 0.1f); // -0.1f to fix rounding
```

Comment:
I'd suggest ~0.75*spellMaxRange

Reply:
Agreed. Otherwise cheesing is quite easy when you can just do 1 step to outrange every time

Reply:
In my experience the chase movement for 40y spell was 30y so 🙈

Reply:
I agree they should go further than max spell range, but 0.75 is a magic number imo. Is there any way we can find the correct chase range 🤔 
I also need to make sure then range doesnt become shorter than min spell range. Bad example: if spell max range is is 40yd and min is 35yd we cannot chase down to 30yd because then spell would become unusable. The solution here is simple but I wonder if there are more edge cases to be aware of - there probably is 😛

Reply:
Yes, we can simply use std::max(spellMaxRange*0.75, spellMinRange), but its back to the question if this is correct behaviour. Im fine with it personally.

Reply:
I personally have no idea. Might be able to do some testing tonight. What Gultask said made sense to me though

Reply:
Good on you for wanting proof though. I agree with that notion